### PR TITLE
autoDetectCORS option for proxy config

### PR DESCRIPTION
**Pull request by mbarto**
## Description
Optional configuration to introduce autodetect of CORS on services used by MapStore2.
If enabled, when contacting any service, MS2 will try to contact it directly, on error will fallback to using the proxy (and continue to use the proxy for following requests).

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
CORS is enabled only if an host is listed in useCORS configuration option.

**What is the new behavior?**
CORS is autodetected and proxy is only used as a fallback.

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/libs/ajax.js [hunk 1]
```diff
@@ -31,6 +32,8 @@ function addHeaderToAxiosConfig(axiosConfig, headerName, headerValue) {
     axiosConfig.headers = assign({}, axiosConfig.headers, {[headerName]: headerValue});
 }
 
+const corsDisabled = [];
```
Comment (offtherailz) [inline web/client/libs/ajax.js]: I'm afraid this is static. Maybe you meant to make it configurable like `proxyConfig.useCORS` array ?

Comment (mbarto) [reply web/client/libs/ajax.js]: No, it's not configurable, it records urls that do not support CORS so the test is not repeated and proxy is used for those.

**Comment by mbarto** [comment]
@offtherailz I probably forgot to commit package.json, I have to check again...

File: web/client/libs/ajax.js (hunk 1 above)
Comment (offtherailz) [reply web/client/libs/ajax.js]: ok, sorry, I didn't get it. So... fine. No problem :smile:

**Pull request merged**